src/idmap.rs

```diff
@@ -17,9 +17,16 @@ use core::ptr::NonNull;
 
 /// Identity mapping, where every virtual address is either unmapped or mapped to the identical IPA.
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub struct IdTranslation;
+pub struct IdTranslation {
+    translation_regime: TranslationRegime,
```

Comment:
If the `TranslationRegime` is needed for any kind of mapping, then why not store it in the `Mapping` instead?